minHeap/main.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"fmt"
+	"math"
+)
+
+/*
+Operations on Min Heap:
+
+getMin(): It returns the root element of Min Heap. O(1)
+
+extractMin(): Removes the minimum element from MinHeap.
+			- This is done by swapping the idx 0 with lastIdx
+			- Then dropping the last idx ( [:len(arr)-1] )
+			- then heapify top-down as the top is now out-of-place
+
+insert(): We add a new key at the end of the tree.
+		- This is done by appending a new element to the back of the array
+		- Then heapify bottomUp because the inserted item could be out of place when compared to its parent
+*/
+
+func main() {
+	mn := NewMinHeap()
+	for i := 0; i < 10; i++ {
+		mn.Insert(i)
+	}
+	for len(mn.items) != 0 {
+		fmt.Println(mn.Extract())
+	}
+}
+
+type MinHeap struct {
+	items []int
+}
```

Comment:
All of this manual implementation can be replaced by [container/heap package ( by implementing their Interface )](https://pkg.go.dev/container/heap#Interface)